### PR TITLE
Adds github workflow for Go Tests

### PR DESCRIPTION
**Pull request by townsymush**

### CODE REVIEW

File: .github/workflows/go-tests.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      - "hotfix/**"
+name: Golang Tests
+jobs:
+  checks:
+    name: run
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Feature Branch
+        uses: actions/checkout@v1
+      - name: run product tests
```
Comment (mikezange) [inline .github/workflows/go-tests.yml:15]: product?

File: .github/workflows/go-tests.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+on:
+  push:
+    branches:
+      - "feature/**"
+      - "bugfix/**"
+      - "hotfix/**"
+name: Golang Tests
+jobs:
```
Comment (mikezange) [inline .github/workflows/go-tests.yml:8]: might be a bit cleaner to do something like this:

```
jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        go: ['1.14.1', '1.13.9', '1.12' , '1.11']
    name: Go ${{ matrix.go }} Test
    steps:
      - uses: actions/checkout@v2
      - name: Setup go
        uses: actions/setup-go@v1
        with:
          go-version: ${{ matrix.go }}
      - run: go test ./...
```

Comment (mikezange) [reply .github/workflows/go-tests.yml:8]: will test multiple versions then 👍

Comment (townsymush) [reply .github/workflows/go-tests.yml:8]: Ahh nice, good idea

**Pull request merged**